modules/dokkatoo-plugin/src/main/kotlin/internal/KotlinNativeDistributionAccessor.kt

```diff
@@ -0,0 +1,39 @@
+@file:Suppress("INVISIBLE_REFERENCE")
+package dev.adamko.dokkatoo.internal
+
+import java.io.File
+import org.gradle.api.Project
+import org.jetbrains.kotlin.commonizer.KonanDistribution
+import org.jetbrains.kotlin.commonizer.platformLibsDir
+import org.jetbrains.kotlin.commonizer.stdlib
+import org.jetbrains.kotlin.compilerRunner.konanHome
+import org.jetbrains.kotlin.konan.target.KonanTarget
+
+/**
+ * Provides access to the Kotlin/Native distribution components:
+ * * [stdlibDir] -- stdlib directory
+ * * [platformDependencies] -- list of directories to platform dependencies
+ *
+ * It uses Kotlin Gradle Plugin API that is guaranteed to be present in:
+ *  1.5 <= kotlinVersion <= 1.9
+ *
+ * It should not be used with Kotlin versions later than 1.9
+ */
+internal class KotlinNativeDistributionAccessor(
+  project: Project
+) {
+  private val konanDistribution = KonanDistribution(
+    @Suppress("INVISIBLE_MEMBER")
+    project.konanHome
+  )
```

Comment:
This workaround is only for <2.0.0, and the API for Kotlin versions is basically fixed.

I'll ask on YouTrack if this can be made public (would be useful for other projects, e.g. to access to `bin/run_konan`).

Reply:
I updated this to use `extensions.extraProperties`, which apparently is needed by the IDE plugin 

https://github.com/JetBrains/kotlin/blob/v1.9.0/libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/targets/native/KotlinNativeTargetPreset.kt#L35-L38